content/blog/2021/06/2021-06-17-student-luminy-project-security.adoc

```diff
@@ -0,0 +1,101 @@
+title: "Four students and their master project in Jenkins security"
+tags:
+- plugins
+- security
+- jenkins
+authors: 
+- jphilip
+- kguerroudj
+- qparra
+- mheyries
+opengraph:
+  image: /images/post-images/2020/08-external-fingerprint-storage-phase-3/opengraph.png
+---
+
+=== Context
+
+Jenkins is the world’s leading open-source automation server used by companies large and small around the globe to implement continuous integration and continuous delivery.
+As such, it is critical that the open source plugins that constitute an integral part of its use as a CI/CD solution don’t expose the systems they are used on to any security risks and vulnerabilities.
+It is in that context that we worked as an audit/code review team to track and report such flaws and problematic practices.
+
+We worked in collaboration with Jenkins Security team member Wadeck Follonier, as part of an end-of-study project during our last year of an IT and cybersecurity Master’s Degree in the university of Aix-Marseille.
+
+
+=== Project Goals
+
+The main goal of this project was to allow us to discover and work on diverse security vulnerabilities in the context of a widely-used software solution, and in order to achieve that we had to separate the project in multiple other goals:
+
+* Learning about some of the most common vulnerabilities and the form in which they can be found on a Jenkins Instance and its plugins.
+
+* Performing code review and technical audit on the application, and compiling our results as part of a security team.
+
+* Reporting our findings to the Jenkins team and the plugin maintainers, while sometimes helping the latter to correct these vulnerabilities.
+
+
+=== Knowledge Sharing
+
+At the beginning of the project, we set up communication channels with our mentor through Slack and Google Meets, and agreed to schedule weekly sessions with him.
+The goal of these sessions was both to teach us more about the functionalities of Jenkins and the types of vulnerabilities we would encounter, and to allow us to ask more specific questions regarding our findings.
+
+Thanks to our mentor developing a mock plugin compiling a variety of classic vulnerabilities and several of their implementations, including SSRFs, XSSs, and XXEs, we have been able to learn through practice.
+It allowed us to analyze the context of the code and the different ways the Jelly framework can be used to display information, expanding our payload options and giving us a clearer view of the patterns to look for during our code reviews.
+
+We have also had the opportunity to learn about the process used to report the vulnerabilities to the maintainers through Jira issues, and some ways we could correct them or provide steps to do so.
+
+
+=== Searching and compiling
+
+At first, we decided to work as a pretty loose team, with each member working on a different plugin and regrouping our findings to confirm or infirm them, while staying in constant contact to ask each other questions.
```

Comment:
I've not seen the word "[infirm](https://www.merriam-webster.com/dictionary/infirm)" used to describe the rejection of a hypothesis.  Possibly means "reject"?
```suggestion
At first, we decided to work as a pretty loose team, with each member working on a different plugin and regrouping our findings to confirm or reject them, while staying in constant contact to ask each other questions.
```